nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/test/java/org/nuxeo/ecm/platform/rendition/service/TestStoredRenditionManager.java

```diff
@@ -74,4 +75,44 @@ protected DocumentModel createBlobDoc(String typeName) {
         return session.createDocument(file);
     }
 
+    @Test
+    public void testStoredRenditionsCleanup() {
+        DocumentModel file1 = createBlobDoc("File");
+        DocumentModel file2 = createBlobDoc("File");
+
+        Rendition ren1 = renditionService.getRendition(file1, "dummyRendition", true);
+        assertNotNull(ren1);
+        DocumentModel renditionDoc1 = ren1.getHostDocument();
+        assertTrue(renditionDoc1.isVersion());
+        DocumentRef liveRenditionRef1 = new IdRef(renditionDoc1.getSourceId());
+        assertTrue(session.exists(liveRenditionRef1));
+
+        Rendition ren2 = renditionService.getRendition(file2, "dummyRendition", true);
+        assertNotNull(ren2);
+        DocumentModel renditionDoc2 = ren2.getHostDocument();
+        assertTrue(renditionDoc2.isVersion());
+        DocumentRef liveRenditionRef2 = new IdRef(renditionDoc2.getSourceId());
+        assertTrue(session.exists(liveRenditionRef2));
+
+        // remove the first document (and versions)
+        session.removeDocument(file1.getRef());
+        eventService.waitForAsyncCompletion();
+
+        TransactionHelper.commitOrRollbackTransaction();
+        TransactionHelper.startTransaction();
```

Comment:
This can be replaced by:

    @Inject
    TransactionalFeature txFeature;

    txFeature.nextTransaction();